blueprints/incredible-squaring/src/main.rs

```diff
@@ -33,18 +36,21 @@ trait GadgetRunner {
     async fn run(&self) -> Result<()>;
 }
 
-struct TangleGadgetRunner {
-    env: gadget_sdk::env::GadgetConfiguration,
+struct TangleGadgetRunner<R: lock_api::RawRwLock> {
+    env: gadget_sdk::env::GadgetConfiguration<R>,
 }
 
 #[async_trait::async_trait]
-impl GadgetRunner for TangleGadgetRunner {
+impl GadgetRunner for TangleGadgetRunner<dyn lock_api::RawRwLock<GuardMarker = GuardSend>> {
```

Comment:
Hmm this seems kinda messy, @Serial-ATA any suggestions to improve how this will integrate with our stack? I do want to hide some of these stuff as best we can when building blueprints. I think instead of `dyn ...` we should just add the trait to the `impl<L: lock_api::RawRwLock<...>>` ... TangleGadgetRunner<L>` if we need it like this @Tjemmmic

Reply:
We could put this RwLock as a type inside the `GadgetEnvironment` trait (which is about to be renamed, as this name is semantically misleading with respect to the new recent uses of an "environment"). We can put every type inside a single spot, allowing us to easily organize and change abstractions at varying levels of the stack.